solr/src/main/java/com/yahoo/ycsb/db/SolrDBClient.java

```diff
@@ -0,0 +1,339 @@
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrInputDocument;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+
+/**
+ * Solr client for YCSB framework.
+ *
+ * <p>
+ * Default properties to set:
+ * </p>
+ * <ul>
+ * <li>table = ycsb
+ * <li>solr.cloud = false
+ * <li>solr.host.list
+ * <li>solr.commit.within.time
+ * <li>solr.batch.mode
+ * <li>zookeeper.host.list
+ * </ul>
+ *
+ * @author Gerard Haughian
+ *
+ */
+public class SolrDBClient extends DB {
+
+  public static final String DEFAULT_COLLECTION_NAME = "ycsb";
+  public static final String DEFAULT_CLOUD_MODE = "false";
+  public static final String DEFAULT_BATCH_MODE = "false";
+  public static final String DEFAULT_REMOTE_ZOOKEEPER_HOST = "localhost:2181";
+  public static final String DEFAULT_REMOTE_SOLR_HOST = "localhost:8983";
+  public static final String DEFAULT_COMMIT_WITHIN_TIME = "1000";
+
+  private SolrClient client;
+  private Integer commitTime;
+  private Boolean batchMode;
+
+  /**
+   * Initialize any state for this DB. Called once per DB instance; there is one DB instance per
+   * client thread.
+   */
+  @Override
+  public void init() throws DBException {
+    Properties props = getProperties();
+    // String collectionName = props.getProperty("table", DEFAULT_COLLECTION_NAME);
+    // TO-DO: Add a check that the collection exists
+    commitTime = Integer
+        .parseInt(props.getProperty("solr.commit.within.time", DEFAULT_COMMIT_WITHIN_TIME));
+    batchMode = Boolean.parseBoolean(props.getProperty("solr.batch.mode", DEFAULT_BATCH_MODE));
+
+    // Check if Solr cluster is running in SolrCloud or Stand-alone mode
+    Boolean cloudMode = Boolean.parseBoolean(props.getProperty("solr.cloud", DEFAULT_CLOUD_MODE));
+    System.out.println("Solr Cloud Mode = " + cloudMode);
+    if (cloudMode) {
+      System.out.println("Solr Zookeeper Remote Hosts = "
+          + props.getProperty("zookeeper.host.list", DEFAULT_REMOTE_ZOOKEEPER_HOST));
+      client = new CloudSolrClient(
+          props.getProperty("zookeeper.host.list", DEFAULT_REMOTE_ZOOKEEPER_HOST));
+    } else { 
+      client = new HttpSolrClient(
+          "http://" + props.getProperty("solr.host.list", DEFAULT_REMOTE_SOLR_HOST) + "/solr");
```

Comment:
Might make sense to have the property be the solr base url which includes the http:// and /solr instead of appending to make the base url. These could conceivably (unlikely though) be different for clusters (ie: https).